src/modules/fancyzones/lib/Zone.cpp

```diff
@@ -81,6 +88,15 @@ void Zone::SizeWindowToZone(HWND window, HWND zoneWindow) noexcept
     if (GetMonitorInfoW(MonitorFromWindow(zoneWindow, MONITOR_DEFAULTTONEAREST), &mi))
     {
         OffsetRect(&zoneRect, mi.rcMonitor.left - mi.rcWork.left, mi.rcMonitor.top - mi.rcWork.top);
+        if (accountForUnawareness)
+        {
+            zoneRect.left = max(mi.rcMonitor.left, zoneRect.left);
+            zoneRect.right = min(mi.rcMonitor.right, zoneRect.right);
+            // Vertical window leaking ~seems~ not to trigger unwanted window resize, so we don't crop it yet 
```

Comment:
Did you try to place one screen on top of the other while putting the taskbar on the edge of screen?

Reply:
yep, I did exactly that.

Reply:
Now that you've mentioned it, I've tried another combination which still triggers the bug even with this code uncommented: you need to have one screen on top of the other with a different scaling and both taskbars enclosing them. I'll investigate it now.

Reply:
@enricogior fixed